### PR TITLE
[FIX] SAM/BAM IO now supports reading padded alignments (cigar op P).

### PR DESCRIPTION
**Pull request by smehringer**
instead of adding a new test I just changed the verbose input/output to have a padded alignment for `read3`

Blocks #1023

### CODE REVIEW

**Comment by rrahn** [comment]
What’s with the code coverage? Something that is not testable? Or false negative?

**Comment by smehringer** [comment]
I added an extra test and made BAM throw on unknown cigar ops.

File: test/unit/io/alignment_file/format_bam_test.cpp [hunk 1]
```diff
@@ -390,6 +390,39 @@ TEST_F(bam_format, wrong_char_as_tag_identifier)
...
+            '\x02', '\x02', '\x03', '\x41', '\x53', '\x43', '\x02', '\x4E', '\x4D', '\x43', '\x07'
+        };
+
+        std::istringstream stream{wrong_char_in_tag};
+        detail::alignment_file_input_format<format_bam> format{};
+
+        EXPECT_THROW(format.read(stream, input_options, this->ref_sequences, this->header, seq, std::ignore,
+                                    std::ignore, std::ignore, std::ignore, std::ignore, std::ignore, alignment, std::ignore,
+                                    std::ignore, std::ignore, tag_dict, std::ignore, std::ignore),
+                    format_error);
```
Comment (rrahn) [inline test/unit/io/alignment_file/format_bam_test.cpp]: Indentation 👀

Comment (smehringer) [reply test/unit/io/alignment_file/format_bam_test.cpp]: :sob:

**Pull request merged**